### PR TITLE
Show the progress interactively with progress bar and in-flight requests (--progress)

### PR DESCRIPTION
**Pull request by xiaochuanyu**
This PR addresses #4.

Note that I'm somewhat new to Rust so I would really appreciate feedback.
One part of this PR that felt a little off to me was having to add lifetime annotation to `Checker`.
Let me know if there is a better way.
capture

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,10 +1,13 @@
+#![type_length_limit = "7912782"]
```
Comment (xiaochuanyu) [inline src/main.rs:1]: This was added due to the compile issue #6.

Comment (mre) [reply src/main.rs:1]: Guess we can remove that setting now that you fixed #6.

**Comment by mre** [comment]
Removed the workaround for the type_length_limit. Fingers crossed that it compiles now. 🤞

**Comment by mre** [comment]
Yay! 🥳

**Comment by mre** [comment]
Thanks for the contribution @xiaochuanyu. Mentioned you in the `README.md` now. 😊

**Comment by mre** [comment]
Forgot to ask: @xiaochuanyu can I add your gif to the `README.md`?

**Pull request merged**